zrml/prediction-markets/src/lib.rs

```diff
@@ -1720,6 +1725,56 @@ mod pallet {
             Ok(Some(weight).into())
         }
 
+        /// Allows the manual closing for "broken" markets.
+        /// A market is "broken", if an unexpected chain stall happened
+        /// and the auto close was scheduled during this time.
+        ///
+        /// # Weight
+        ///
+        /// Complexity: `O(n)`,
+        /// and `n` is the number of market ids,
+        /// which close at the same time as the specified market.
+        #[pallet::call_index(21)]
+        #[pallet::weight(T::WeightInfo::manually_close_market(CacheSize::get()))]
+        #[transactional]
+        pub fn manually_close_market(
+            origin: OriginFor<T>,
+            #[pallet::compact] market_id: MarketIdOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            ensure_signed(origin)?;
+
+            let market = zrml_market_commons::Pallet::<T>::market(&market_id)?;
+            let now = zrml_market_commons::Pallet::<T>::now();
+            let range = match &market.period {
+                MarketPeriod::Block(_) => {
+                    return Err(Error::<T>::NotAllowedForBlockBasedMarkets.into());
+                }
+                MarketPeriod::Timestamp(ref range) => range,
+            };
+
+            let close_ids_len = if range.end <= now {
+                let range_end_time_frame = Self::calculate_time_frame_of_moment(range.end);
+                let close_ids_len = MarketIdsPerCloseTimeFrame::<T>::try_mutate(
+                    range_end_time_frame,
+                    |ids| -> Result<u32, DispatchError> {
+                        let ids_len = ids.len() as u32;
+                        let position = ids
+                            .iter()
+                            .position(|i| i == &market_id)
+                            .ok_or(Error::<T>::MarketNotInCloseTimeFrameList)?;
+                        let _ = ids.swap_remove(position);
+                        Ok(ids_len)
+                    },
+                )?;
+                Self::on_market_close(&market_id, market)?;
+                close_ids_len
+            } else {
+                return Err(Error::<T>::MarketPeriodNotStartedYet.into());
```

Comment:
This sounds like the wrong error.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/1160/commits/f41f435b1790ba08c2e26b64301223d7ecc437d6